advanced/creating-an-icon-pack.md

```diff
@@ -0,0 +1,120 @@
+2. Open a terminal or command prompt, navigate to the `oxide-icon-pack-template` directory.
+3. Install the project dependencies by executing:
+
+    ```sh
+    $ npm install
+    ```
+4. When prompted, enter a name for the icon pack. The icon pack name should only contain:
+
+* Numbers.
+* Letters.
+* Hyphens ( `-` ).
+* Underscores ( `_` ).
+
+The icon pack name will be used with the [icons]({{site.baseurl}}/configure/editor-appearance/#icons) option to apply the icons in TinyMCE.
+
+### Add the SVG files
+
+Each SVG files placed in `/src/svg` will be converted to an icon. The file names of the SVG files are used to set the icon identifier used by TinyMCE.
+
+For example: `bold.svg` will have the identifier `bold`. Such as:
+
+```js
+tinymce.init({
+  selector: '#tiny_custom_button',
+  toolbar: 'myButton',
+  icons: 'my_icon_pack',
+  setup: function (editor) {
+    editor.ui.registry.addButton('myButton', {
+      icon: 'bold',    // the 'bold' icon  created from 'bold.svg'
+      onAction: function (_) {
+        editor.insertContent('&nbsp;<strong>It\'s my icon button!</strong>&nbsp;');
+      }
+    });
+  }
+});
+```
+
+For a list of the icon identifiers, see: [Editor icon identifiers]({{site.baseurl}}/advanced/editor-icon-identifiers/).
+
+TinyMCE does not resize the SVGs provided, relying on the size defined in the SVG. This allows icons of different sizes to be used in the editor. The Toolbar button sizes are independent of the icon sizes. To change button sizes, a [custom skin]({{site.baseurl}}/advanced/creating-a-skin/) is required.
+
+> **Note**: Input SVGs must be shapes, not strokes. SVG files containing strokes will not render correctly. If the input files contain strokes, use a graphics program to convert the strokes into shapes.
+
+### Build the icon pack
+
+To build the icon pack using Gulp:
+
+1. Open a terminal or command prompt and navigate to the root directory of the icon pack (such as: `oxide-icon-pack-template/`).
+2. Build the icon pack by executing the `gulp` command:
+
+    ```sh
+    $ gulp
+    ```
+    A `dist/` directory containing the icon pack will be created.
+3. Using a web browser, open `dist/html/icons.html` to preview the icons.
+
+#### Troubleshooting information for building icon packs
+The SVG files are optimized during the build process with [SVGO](https://github.com/svg/svgo). The optimization can result in distorted graphics due to rounding errors. The graphics may be fixed by providing new SVGO options. To change the SVGO options used:
+
+1. Using a text editor, open `gulpfile.js`.
+2. Add the `svgo` option to the `iconPackager` function, such as:
+
+    ```js
+    iconPackager({
+      name: 'my-icon-pack',
+      svgo: { floatPrecision: 3 } //Increase the rounding precision
+    })
+    ```
+All user defined options, including SVGO options, will merge with the default options. For information on SVGO options, see: [SVGO on GitHub](https://github.com/svg/svgo).
+
+## Deploying an icon pack
```

Comment:
Does this need information about how to deploy a custom icon pack on cloud (ie via the `icons_url` option)? As the contents here only apply to self-hosted as far as I can see.

Reply:
True. Updated and added content from `icons_url`

Reply:
Looks good to me thanks! :+1: